src/interrogate/badge_gen.py

```diff
@@ -16,9 +17,13 @@
 except ImportError:  # pragma: no cover
     cairosvg = None
 
+from interrogate.coverage import InterrogateResults
 
-DEFAULT_FILENAME = "interrogate_badge"
-COLORS = {
+
+NumberType = int | float
```

Comment:
This causes CI to fail as `int | float` is actually executed at runtime (since it's not an annotation). In this case you still have to use the <3.10 syntax:
```suggestion
NumberType = Union[int, float]
```
Or, better yet, don't introduce this type alias at all and simply use `int | float` everywhere (it's just one more character :wink:)